ONNX-based Higgs to bb Interaction Network Tagger

jmduarte opened the pull request:
#### PR description:

This PR implements the Higgs to bb Interaction Network tagger into CMSSW. This model was developed originally using open simulation (https://arxiv.org/abs/1909.12285). The version here corresponds to this original version. A re-training with mass-varying samples will be performed in the future. This model has been presented in several BTV meetings (e.g. https://indico.cern.ch/event/833434/contributions/3495704/) and JMAR meetings (e.g. https://indico.cern.ch/event/856106/contributions/3613884/).

The model was developed in PyTorch and converted to ONNX for use in CMSSW.

Execution time with 4 threads for 2017 JetHT events (and compared to other Deep taggers for reference) is:
```
TimeReport ---------- Module Summary ---[Real sec]----
TimeReport   0.000274     0.000274     0.000274  pfDeepDoubleXTagInfosSlimmedAK8DeepTags
TimeReport   0.000739     0.000739     0.000739  pfDeepBoostedJetTagInfosSlimmedAK8DeepTags
TimeReport   0.000738     0.000738     0.000738  pfParticleNetTagInfosSlimmedAK8DeepTags
TimeReport   0.000633     0.000633     0.000633  pfHiggsInteractionNetTagInfosSlimmedAK8DeepTags
TimeReport   0.000512     0.000512     0.000512  pfDeepDoubleBvLJetTagsSlimmedAK8DeepTags
TimeReport   0.004054     0.004054     0.004054  pfDeepBoostedJetTagsSlimmedAK8DeepTags
TimeReport   0.104109     0.104109     0.104109  pfParticleNetJetTagsSlimmedAK8DeepTags
TimeReport   0.007841     0.007841     0.007841  pfHiggsInteractionNetTagsSlimmedAK8DeepTags
```

Needs cms-data/RecoBTag-Combined#30
#### PR validation:

This PR was verified with the training framework and shows the same results.

![probHbbprobQCD](https://user-images.githubusercontent.com/4932543/83477952-67d80680-a449-11ea-823a-bce802ce6d87.png)

jmduarte committed:
add Higgs IN tagger

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30072/15799

- This PR adds an extra 60KB to repository

- Found files with invalid states:
  - RecoBTag/ONNXRuntime/plugins/HiggsINONNXJetTagsProducer.cc:
    - Added: 1ac1ca806cdce604f0e731c5ea81feab1b5ee409
    - Deleted: ae33c31fb8014d880fc46484d4f0bab91a527519
  - RecoBTag/ONNXRuntime/python/pfHiggsIN_cff.py:
    - Added: 1ac1ca806cdce604f0e731c5ea81feab1b5ee409
    - Deleted: ae33c31fb8014d880fc46484d4f0bab91a527519
  - RecoBTag/ONNXRuntime/python/Parameters/HiggsIN/V00/pfHiggsINPreprocessParams_cfi.py:
    - Added: 1ac1ca806cdce604f0e731c5ea81feab1b5ee409
    - Deleted: ae33c31fb8014d880fc46484d4f0bab91a527519
  - RecoBTag/FeatureTools/plugins/HiggsINTagInfoProducer.cc:
    - Added: 1ac1ca806cdce604f0e731c5ea81feab1b5ee409
    - Deleted: ae33c31fb8014d880fc46484d4f0bab91a527519
  - DataFormats/BTauReco/interface/HiggsINTagInfo.h:
    - Added: 1ac1ca806cdce604f0e731c5ea81feab1b5ee409
    - Deleted: ae33c31fb8014d880fc46484d4f0bab91a527519
  - DataFormats/BTauReco/interface/HiggsINFeatures.h:
    - Added: 1ac1ca806cdce604f0e731c5ea81feab1b5ee409
    - Deleted: ae33c31fb8014d880fc46484d4f0bab91a527519


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/BTauReco/src/classes_def.xml  modified in PR(s): #30016

cmsbuild commented:
A new Pull Request was created by  @jmduarte (Javier Duarte) for master.

It involves the following packages:

DataFormats/BTauReco
PhysicsTools/PatAlgos
RecoBTag/Configuration
RecoBTag/FeatureTools
RecoBTag/ONNXRuntime


@perrotta, @cmsbuild, @santocch, @slava77 can you please review it and eventually sign? Thanks.
@rappoccio, @gouskos, @hatakeyamak, @emilbols, @peruzzim, @seemasharmafnal, @mmarionncern, @ahinzmann, @smoortga, @jdolen, @hqucms, @ferencek, @rovere, @jdamgov, @nhanvtran, @gkasieczka, @schoef, @mariadalfonso, @clelange, @riga, @JyothsnaKomaragiri, @mverzett, @gpetruc, @andrzejnovak this is something you requested to watch as well.
@silviodonato, @dpiparo you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30072/15800

- This PR adds an extra 56KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/BTauReco/src/classes_def.xml  modified in PR(s): #30016


Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30072/15800/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30072/15800/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

jmduarte committed:
code checks

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/RecoBTag-Combined#30

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30072/15812

- This PR adds an extra 56KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/BTauReco/src/classes_def.xml  modified in PR(s): #30016

cmsbuild commented:
Pull request #30072 was updated. @perrotta, @santocch, @slava77 can you please check and sign again.

slava77 commented:
test parameters
- pull_request = cms-data/RecoBTag-Combined#30

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/RecoBTag-Combined#30

cmsbuild commented:
-1

Tested at: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d

CMSSW: CMSSW_11_2_X_2020-06-02-1100
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/6746/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```
 See details on the summary page.

cmsbuild commented:
Comparison not run due to Build errors/Fireworks only changes/No short matrix requested (RelVals and Igprof tests were also skipped)

jmduarte committed:
fix clang warning; no copy

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30072/15820

- This PR adds an extra 56KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/BTauReco/src/classes_def.xml  modified in PR(s): #30016

cmsbuild commented:
Pull request #30072 was updated. @perrotta, @cmsbuild, @santocch, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/RecoBTag-Combined#30

cmsbuild commented:
+1
Tested at: 1dee5d6b845f3d70d5f6ebf3f352eedeb10f63f2
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/6789/summary.html
CMSSW: CMSSW_11_2_X_2020-06-02-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/6789/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3053 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2783666
 - DQMHistoTests: Total failures: 51
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2783565
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

jmduarte commented:
@slava77 is this waiting for a follow-up from me?

slava77 commented:
> @slava77 is this waiting for a follow-up from me?

no, it's pending review (from me)
I plan for this week

slava77 commented on DataFormats/BTauReco/interface/HiggsInteractionNetFeatures.h:
```diff
@@ -0,0 +1,62 @@
+
+namespace btagbtvdeep {
+
+  class HiggsInteractionNetFeatures {
```
this looks identical to ```DeepBoostedJetFeatures```; please remove this class and move to ```DeepBoostedJetFeatures```

jmduarte replied:
Removed at 858b30df397. We also removed `HiggsInteractionNetTagInfos` because we could reuse `DeepBoostedJetTagInfos`. Let me know if that's appropriate.

jmduarte committed:
rm HiggsInteractionNetFeatures and HiggsInteractionNetTagInfoå

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30072/16100

- This PR adds an extra 60KB to repository

- Found files with invalid states:
  - DataFormats/BTauReco/interface/HiggsInteractionNetFeatures.h:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d
    - Deleted: 858b30df39705ef42d7286f474ce5579d1d202ea
  - DataFormats/BTauReco/interface/HiggsInteractionNetTagInfo.h:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Deleted: 858b30df39705ef42d7286f474ce5579d1d202ea


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  PhysicsTools/PatAlgos/python/recoLayer0/bTagging_cff.py  modified in PR(s): #30016
  - File  PhysicsTools/PatAlgos/python/slimming/applyDeepBtagging_cff.py  modified in PR(s): #30016

cmsbuild commented:
Pull request #30072 was updated. @perrotta, @cmsbuild, @santocch, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/RecoBTag-Combined#30

cmsbuild commented:
+1
Tested at: 858b30df39705ef42d7286f474ce5579d1d202ea
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/7021/summary.html
CMSSW: CMSSW_11_2_X_2020-06-12-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/7021/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3260 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2778811
 - DQMHistoTests: Total failures: 4
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2778757
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

slava77 commented:
> Execution time with 4 threads for 2017 JetHT events (and compared to other Deep taggers for reference) is:

in which release were the timing tests done?

jmduarte commented:
Hi @slava77, I did it in  CMSSW_11_2_X_2020-05-31-2300 IB. If there is a more appropriate release or data sample, let me know.

slava77 commented:
> Hi @slava77, I did it in `CMSSW_11_2_X_2020-05-31-2300` IB. If there is a more appropriate release or data sample, let me know.

this is good; thanks.

slava77 commented on RecoBTag/FeatureTools/plugins/HiggsInteractionNetTagInfoProducer.cc:
```diff
@@ -0,0 +1,353 @@
+    }
+    TrackInfoBuilder trkinfo(track_builder_);
+    trkinfo.buildTrackInfo(&(*cand), jet_dir, jet_ref_track_dir, *pv_);
+    fts.fill("cpf_btagEtaRel", catch_infs_and_bound(trkinfo.getTrackEtaRel(), 0, -5, 15));
+    fts.fill("cpf_btagPtRatio", catch_infs_and_bound(trkinfo.getTrackPtRatio(), 0, -1, 10));
+    fts.fill("cpf_btagPParRatio", catch_infs_and_bound(trkinfo.getTrackPParRatio(), 0, -10, 100));
```
The choice of negative values seems a bit arbitrary.
are the negative bounds even possible?

jmduarte replied:
These choices were made just to match the original ntupler used for training (https://github.com/DeepDoubleB/DNNTuplesAK8/blob/opendata_80X/BTagHelpers/src/TrackInfoBuilder.cc#L56-L70). 
This results in 100% agreement for the tagger output. 

Changing these (here and elsewhere) will have some (possibly small) effect on the tagger output and possibly break the 100% agreement.

I can check how far we are from 100% and see if we can live with it (i.e. if tagger performance won't be greatly affected).

slava77 replied:
my comment implied that all the "Rel" are with respect to the jet axis; if the cands are from the jet, how can any of the values be negative?
Hence, the negative values as lower bounds are just random.

slava77 commented on RecoBTag/FeatureTools/plugins/HiggsInteractionNetTagInfoProducer.cc:
```diff
@@ -0,0 +1,353 @@
+    fts.fill("cpf_btagPParRatio", catch_infs_and_bound(trkinfo.getTrackPParRatio(), 0, -10, 100));
+    fts.fill("cpf_btagSip2dVal", catch_infs_and_bound(trkinfo.getTrackSip2dVal(), 0, -1, 70));
+    fts.fill("cpf_btagSip2dSig", catch_infs_and_bound(trkinfo.getTrackSip2dSig(), 0, -1, 4e4));
+    fts.fill("cpf_btagSip3dVal", catch_infs_and_bound(trkinfo.getTrackSip3dVal(), 0, -1, 1e5));
```
is the upper bound realistic or just a random number?
Wouldn't the same as for Sip2dVal just work?

slava77 commented on RecoBTag/FeatureTools/plugins/HiggsInteractionNetTagInfoProducer.cc:
```diff
@@ -0,0 +1,353 @@
+    fts.fill("cpf_btagSip2dVal", catch_infs_and_bound(trkinfo.getTrackSip2dVal(), 0, -1, 70));
+    fts.fill("cpf_btagSip2dSig", catch_infs_and_bound(trkinfo.getTrackSip2dSig(), 0, -1, 4e4));
+    fts.fill("cpf_btagSip3dVal", catch_infs_and_bound(trkinfo.getTrackSip3dVal(), 0, -1, 1e5));
+    fts.fill("cpf_btagSip3dSig", catch_infs_and_bound(trkinfo.getTrackSip3dSig(), 0, -1, 4e4));
```
individual track significances are clamped at 2e3, would the same value work here?
Again, trying to reduce the entropy of random numbers in the code.

slava77 commented on RecoBTag/FeatureTools/plugins/HiggsInteractionNetTagInfoProducer.cc:
```diff
@@ -0,0 +1,353 @@
+
+    // sv properties
+    fts.fill("sv_ntracks", sv->numberOfDaughters());
+    fts.fill("sv_normchi2", catch_infs_and_bound(sv->vertexNormalizedChi2(), 1000, -1000, 1000));
```
the track lower bound was at -1, is -1e3 any better?

slava77 commented on RecoBTag/ONNXRuntime/plugins/HiggsInteractionNetONNXJetTagsProducer.cc:
```diff
@@ -0,0 +1,214 @@
+
+void HiggsInteractionNetONNXJetTagsProducer::make_inputs(const reco::DeepBoostedJetTagInfo &taginfo) {
+  for (unsigned igroup = 0; igroup < input_names_.size(); ++igroup) {
+    const auto &group_name = input_names_.at(igroup);
+    auto &group_values = data_.at(igroup);
+    const auto &prep_params = prep_info_map_.at(group_name);
```
```suggestion
    const auto &group_name = input_names_[igroup];
    auto &group_values = data_[igroup];
    const auto &prep_params = prep_info_map_[group_name];
```

slava77 reviewed (commented):
- HiggsInteractionNetTagInfoProducer has a very large overlap with DeepBoostedJetTagInfoProducer. Ideally, the DeepBoostedJetTagInfoProducer should be made configurable to be able to produce the taginfos/features for the InteractionNet inference.
At a glance, the leading reason for a still fairly significant difference between the two is in clamping of ranges of some variables (with at least some probably having no effect on either inference performance). Please review the differences towards reusing the same feature filling code.
    - note that DeepBoostedJetONNXJetTagsProducer already supports offsetting and scaling the features; it should be rather simple to extend it to also support the range bounds
- HiggsInteractionNetONNXJetTagsProducer overlaps very significantly in implementation with DeepBoostedJetONNXJetTagsProducer. with only small modifications to DeepBoostedJetONNXJetTagsProducer it should be possible to reuse it for the InteractionNet inference

jmduarte commented on RecoBTag/FeatureTools/plugins/HiggsInteractionNetTagInfoProducer.cc:
```diff
@@ -0,0 +1,353 @@
+                          cand->pt() / jet.pt());
+  }
+
+  std::sort(c_sorted.begin(), c_sorted.end(), btagbtvdeep::SortingClass<reco::CandidatePtr>::compareByABCInv);
```
The only issue I noticed is that when there are more than 60 charged PF candidates (which happens rarely but not super rarely), then since we only take the first 60 in the tag producer, we are actually taking a different set than the ones used to train the algorithm.

slava77 commented:
@jmduarte 
please clarify on the status of the follow up to this PR review.
Thank you.

jmduarte commented:
@slava77 yes, sorry for the delay. I will try to respond to the comments and update the PR today/tomorrow.

jmduarte committed:
reuse DeepBoostedJet Producers

jmduarte commented:
@slava77  In another branch (https://github.com/jmduarte/cmssw/tree/higgs-in-tagger-rebased), I've migrated to re-using `DeepBoostedJetTagInfoProducer` and I am testing the effects of the different ranges and sorting (https://www.dropbox.com/s/4blk783x39e2zaz/TagInfos.pdf?dl=0). I now have agreement for the Interaction Network at the <5% level (though certainly affecting DeepAK8 results at the moment). Once I track down the remaining differences, I plan to update this PR, restoring agreement for DeepAK8 and addressing the remaining comments.

slava77 commented:
> @slava77 In another branch (https://github.com/jmduarte/cmssw/tree/higgs-in-tagger-rebased), I've migrated to re-using `DeepBoostedJetTagInfoProducer` and I am testing the effects of the different ranges and sorting (https://www.dropbox.com/s/4blk783x39e2zaz/TagInfos.pdf?dl=0). I now have agreement for the Interaction Network at the <5% level (though certainly affecting DeepAK8 results at the moment). Once I track down the remaining differences, I plan to update this PR, restoring agreement for DeepAK8 and addressing the remaining comments.

related to DeepBoostedJetTagInfoProducer,
offset and scaling is already something that can be handled by the tag producer at the input step using configuration.
It seems rather straightforward to also add the clamping option, which would allow identical results compared to the initial proposal with (varying) hardcoded offsets and bounds.
(see the review summary https://github.com/cms-sw/cmssw/pull/30072#pullrequestreview-432823540)
@jmduarte did you consider this option?

jmduarte commented:
@slava77, yes that is exactly what I'm doing following your suggestion. A prototype of this is now implemented in https://github.com/jmduarte/cmssw/tree/higgs-in-tagger-rebased, which gets perfect agreement with the training framework for the Interaction Network. Now, I just have to clean it up, ensure consistency for DeepAK8, and push it to this PR.

I should be able to generalize/reuse `DeepBoostedJetONNXJetTagsProducer` as you suggested as well.

cmsbuild commented:
The code-checks are being triggered in jenkins.

jmduarte commented:
hi @slava77, I have made the following changes:

- Added additional configurations for  `DeepBoostedJetTagInfoProducer` and now reused it for both DeepAK8 and HiggsInteractionNetwork
- Added additional configurations for catching infs and clamping per input variable to `DeepBoostedJetONNXJetTagsProducer` and now reused it for both DeepAK8 and HiggsInteractionNetwork

Please let me know what you think. @hqucms should probably take a look at the changes as well, although I've checked locally that DeepAK8 values appear unchanged [1].

For HiggsInteractionNetwork, things agree as well, except for one event in 10k [2]. Actually, this minor disagreement was there before too, and shouldn't affect performance in any meaningful way.


[1] DeepAK8 agreement before and after
![corr_deepak8](https://user-images.githubusercontent.com/4932543/85965986-55cb8400-b973-11ea-8e2d-abe9a33963c7.png)
![res_deepak8](https://user-images.githubusercontent.com/4932543/85966024-6d0a7180-b973-11ea-8674-44c6654769f8.png)

[2] HiggsInteractionNetwork agreement with training framework
![corr (10)](https://user-images.githubusercontent.com/4932543/85966054-84495f00-b973-11ea-8698-ae84291bd532.png)
![res (10)](https://user-images.githubusercontent.com/4932543/85966062-8dd2c700-b973-11ea-8afe-fb35715932aa.png)

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30072/16564

- This PR adds an extra 72KB to repository

- Found files with invalid states:
  - DataFormats/BTauReco/interface/HiggsInteractionNetFeatures.h:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d
    - Deleted: 858b30df39705ef42d7286f474ce5579d1d202ea
  - RecoBTag/ONNXRuntime/plugins/HiggsInteractionNetONNXJetTagsProducer.cc:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d, 858b30df39705ef42d7286f474ce5579d1d202ea
    - Deleted: 9bce4d38a9139531cb3c9fcbc796a99e57156519
  - RecoBTag/FeatureTools/plugins/HiggsInteractionNetTagInfoProducer.cc:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d, 1dee5d6b845f3d70d5f6ebf3f352eedeb10f63f2, 858b30df39705ef42d7286f474ce5579d1d202ea
    - Deleted: 9bce4d38a9139531cb3c9fcbc796a99e57156519
  - DataFormats/BTauReco/interface/HiggsInteractionNetTagInfo.h:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Deleted: 858b30df39705ef42d7286f474ce5579d1d202ea


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  PhysicsTools/PatAlgos/python/recoLayer0/bTagging_cff.py  modified in PR(s): #30016

cmsbuild commented:
Pull request #30072 was updated. @perrotta, @cmsbuild, @santocch, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/RecoBTag-Combined#30

cmsbuild commented:
-1

Tested at: 9bce4d38a9139531cb3c9fcbc796a99e57156519

CMSSW: CMSSW_11_2_X_2020-06-29-1100
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/7484/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
136.8311 step2
<pre>runTheMatrix-results/136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017/step2_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017.log

</pre>136.88811 step2
<pre>runTheMatrix-results/136.88811_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL/step2_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL.log

</pre>136.7611 step2
<pre>runTheMatrix-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM/step2_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM.log

</pre>4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>136.874 step3
<pre>runTheMatrix-results/136.874_RunEGamma2018C+RunEGamma2018C+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Offline_L1TEgDQM+HARVEST2018_L1TEgDQM/step3_RunEGamma2018C+RunEGamma2018C+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Offline_L1TEgDQM+HARVEST2018_L1TEgDQM.log

</pre>136.731 step3
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step3_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>136.793 step3
<pre>runTheMatrix-results/136.793_RunDoubleEG2017C+RunDoubleEG2017C+HLTDR2_2017+RECODR2_2017reHLT_skimDoubleEG_Prompt+HARVEST2017/step3_RunDoubleEG2017C+RunDoubleEG2017C+HLTDR2_2017+RECODR2_2017reHLT_skimDoubleEG_Prompt+HARVEST2017.log

</pre>9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>25.0 step3
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>10024.0 step3
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017+NanoFull_2017/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017+NanoFull_2017.log

</pre>23234.1001 step1
<pre>runTheMatrix-results/23234.1001_TTbar_14TeV+RecoFullGlobal_TestOldDigi_2026D49+HARVESTFullGlobal_2026D49/step1_TTbar_14TeV+RecoFullGlobal_TestOldDigi_2026D49+HARVESTFullGlobal_2026D49.log

</pre>10824.0 step3
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018+NanoFull_2018/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018+NanoFull_2018.log

</pre>10224.0 step3
<pre>runTheMatrix-results/10224.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU+NanoFull_2017PU/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU+NanoFull_2017PU.log

</pre>11634.0 step3
<pre>runTheMatrix-results/11634.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull+DigiFull_2021+RecoFull_2021+HARVESTFull_2021+ALCAFull_2021/step3_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull+DigiFull_2021+RecoFull_2021+HARVESTFull_2021+ALCAFull_2021.log

</pre>12434.0 step3
<pre>runTheMatrix-results/12434.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023/step3_TTbar_14TeV+TTbar_14TeV_TuneCP5_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023.log

</pre>20034.0 step3
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35/step3_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D35+RecoFullGlobal_2026D35+HARVESTFullGlobal_2026D35.log

</pre>20434.0 step3
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D41+RecoFullGlobal_2026D41+HARVESTFullGlobal_2026D41/step3_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D41+RecoFullGlobal_2026D41+HARVESTFullGlobal_2026D41.log

</pre>23234.0 step3
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49/step3_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D49_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D49+RecoFullGlobal_2026D49+HARVESTFullGlobal_2026D49.log

</pre>21234.0 step3
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D44_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D44+RecoFullGlobal_2026D44+HARVESTFullGlobal_2026D44/step3_TTbar_14TeV+TTbar_14TeV_TuneCP5_2026D44_GenSimHLBeamSpotFull14+DigiFullTrigger_2026D44+RecoFullGlobal_2026D44+HARVESTFullGlobal_2026D44.log

</pre>250202.181 step4
<pre>runTheMatrix-results/250202.181_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25/step4_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

slava77 commented:
```
An exception of category 'InvalidArgument' occurred while
   [0] Processing  Event run: 194533 lumi: 329 event: 461583409 stream: 0
   [1] Running path 'dqmofflineOnPAT_1_step'
   [2] Prefetching for module JetAnalyzer/'jetDQMAnalyzerAk8PFPUPPICleanedMiniAOD'
   [3] Prefetching for module PATJetSelector/'slimmedJetsAK8'
   [4] Prefetching for module PATJetUpdater/'updatedPatJetsTransientCorrectedSlimmedAK8DeepTags'
   [5] Calling method for module BoostedJetMXNetJetTagsProducer/'pfParticleNetJetTagsSlimmedAK8DeepTags'
Exception Message:
Cannot find variable info for pfcand_etarel
```
looks related to the PR

jmduarte committed:
factorize PreprocessParams

jmduarte commented:
Hi @slava77,  the issue was that this struct `PreprocessParams` was defined in two places [1] [2] and they needed to be identical or things break. I've now factored this struct out into `deep_helpers.h` in the `btagbtvdeep` namespace, and use it in both producers.

[1] https://github.com/cms-sw/cmssw/blob/4d8bfcb461e8314ed0f38cf3bf19d40433c26ab2/RecoBTag/MXNet/plugins/BoostedJetMXNetJetTagsProducer.cc#L30
[2] https://github.com/cms-sw/cmssw/blob/4d8bfcb461e8314ed0f38cf3bf19d40433c26ab2/RecoBTag/ONNXRuntime/plugins/DeepBoostedJetONNXJetTagsProducer.cc#L26

cmsbuild commented:
The code-checks are being triggered in jenkins.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30072/16598

- This PR adds an extra 32KB to repository

- Found files with invalid states:
  - RecoBTag/ONNXRuntime/plugins/HiggsInteractionNetONNXJetTagsProducer.cc:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d, 858b30df39705ef42d7286f474ce5579d1d202ea
    - Deleted: 9bce4d38a9139531cb3c9fcbc796a99e57156519
  - DataFormats/BTauReco/interface/HiggsInteractionNetFeatures.h:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d
    - Deleted: 858b30df39705ef42d7286f474ce5579d1d202ea
  - RecoBTag/FeatureTools/plugins/HiggsInteractionNetTagInfoProducer.cc:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d, 1dee5d6b845f3d70d5f6ebf3f352eedeb10f63f2, 858b30df39705ef42d7286f474ce5579d1d202ea
    - Deleted: 9bce4d38a9139531cb3c9fcbc796a99e57156519
  - DataFormats/BTauReco/interface/HiggsInteractionNetTagInfo.h:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Deleted: 858b30df39705ef42d7286f474ce5579d1d202ea


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  PhysicsTools/PatAlgos/python/recoLayer0/bTagging_cff.py  modified in PR(s): #30016
  - File  RecoBTag/MXNet/plugins/BoostedJetMXNetJetTagsProducer.cc  modified in PR(s): #29934

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/RecoBTag-Combined#30

cmsbuild commented:
Pull request #30072 was updated. @perrotta, @cmsbuild, @santocch, @slava77 can you please check and sign again.

cmsbuild commented:
+1
Tested at: a3f7eb5fc2de19761b392033c919adf391e37aa4
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/7504/summary.html
CMSSW: CMSSW_11_2_X_2020-06-29-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/7504/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3256 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2778915
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2778864
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

slava77 commented:
I notice that the test summary has errors, which are not reported in this thread 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/7504/summary.html
For "Build rules" the link leads to something undeciferable newer-than-target.log
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/7504/newer-than-target.log

@smuzaffar @makortel 
is this some kind of glitch or is there something to fix for this PR?

jmduarte commented:
Hi @slava77, as far as I can tell it's unrelated to this PR, but I'm also unsure of how to parse these logs.

slava77 commented on RecoBTag/ONNXRuntime/python/pfHiggsInteractionNet_cff.py:
```diff
@@ -0,0 +1,31 @@
+import FWCore.ParameterSet.Config as cms
+
+from RecoBTag.FeatureTools.pfDeepBoostedJetTagInfos_cfi import pfDeepBoostedJetTagInfos as _pfHiggsInteractionNetTagInfos
+from RecoBTag.ONNXRuntime.pfDeepBoostedJetTags_cfi import pfDeepBoostedJetTags as _pfHiggsInteractionNetTags
```
in this context the renaming to ```_pfHiggsInteractionNetTagInfos``` is just an obfuscation, I suggest to use ```pfDeepBoostedJetTagInfos``` and ```pfDeepBoostedJetTags``` directly

jmduarte replied:
For some reason, using them directly results in some conflict when running multiple taggers, with an error like:
```
----- Begin Fatal Exception 02-Jul-2020 22:36:29 PDT-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=DeepBoostedJetONNXJetTagsProducer label='pfDeepBoostedJetTags'
Exception Message:
MissingParameter: Parameter 'input_names' not found.
----- End Fatal Exception -------------------------------------------------
```
I've left the renaming for now (just by putting a `_` in front).

slava77 commented on RecoBTag/ONNXRuntime/python/pfHiggsInteractionNet_cff.py:
```diff
@@ -0,0 +1,31 @@
+
+# modify default parameters for tag infos
+pfHiggsInteractionNetTagInfos = _pfHiggsInteractionNetTagInfos.clone(
+    min_pt_for_track_properties = cms.double(0.95),
```
please drop type specifications for all parameters which already exist.
This is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.

```suggestion
    min_pt_for_track_properties = 0.95,
```
same for other parameters cases in this file

jmduarte replied:
Done.

slava77 reviewed (commented):
@jmduarte 
thank you for the updates following the previous review.
It looks like we are almost there.

makortel commented:
> is this some kind of glitch or is there something to fix for this PR?

Looks like a glitch to me, but better for @smuzaffar to take a look.

jmduarte committed:
respond to slava's review

cmsbuild commented:
The code-checks are being triggered in jenkins.

jmduarte commented on RecoBTag/ONNXRuntime/plugins/DeepBoostedJetONNXJetTagsProducer.cc:
```diff
@@ -239,35 +226,40 @@ std::vector<float> DeepBoostedJetONNXJetTagsProducer::center_norm_pad(const std:
   for (unsigned igroup = 0; igroup < input_names_.size(); ++igroup) {
-    const auto &group_name = input_names_[igroup];
-    auto &group_values = data_[igroup];
+    const auto &group_name = input_names_.at(igroup);
```
Fixed to use `[]`.

jmduarte commented on RecoBTag/MXNet/python/Parameters/ParticleNet/V00/pfParticleNetPreprocessParams_cfi.py:
```diff
@@ -14,106 +14,182 @@
+                norm_factor = cms.double(1.0),
+                replace_value = cms.double(0),
+                lower_bound = cms.double(-5),
+                upper_bound = cms.double(5),
```
This hasn't been discussed specifically (except in the context of merging the tag info producers in general). These parameters could be read in from the model file (or an auxiliary data file) in a follow-up PR.

jmduarte commented on RecoBTag/MXNet/plugins/BoostedJetMXNetJetTagsProducer.cc:
```diff
@@ -248,35 +236,40 @@ std::vector<float> BoostedJetMXNetJetTagsProducer::center_norm_pad(const std::ve
   for (unsigned igroup = 0; igroup < input_names_.size(); ++igroup) {
-    const auto &group_name = input_names_[igroup];
-    auto &group_values = data_[igroup];
+    const auto &group_name = input_names_.at(igroup);
```
Fixed to `[]`.

jmduarte commented on RecoBTag/FeatureTools/interface/deep_helpers.h:
```diff
@@ -74,5 +74,38 @@ namespace btagbtvdeep {
+            upper_bound(upper_bound) {}
+      float center = 0;
+      float norm_factor = 1;
+      float replace_value = 0;
```
Renamed to `replace_inf_value`.

jmduarte commented on RecoBTag/FeatureTools/plugins/DeepBoostedJetTagInfoProducer.cc:
```diff
@@ -290,6 +308,12 @@ void DeepBoostedJetTagInfoProducer::fillParticleFeatures(DeepBoostedJetFeatures
+      continue;
+    } else if (!include_neutrals_ && reco_cand && !useTrackProperties(reco_cand)) {
+      continue;
+    }
```
Changed as you suggest.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30072/16726

- This PR adds an extra 132KB to repository

- Found files with invalid states:
  - RecoBTag/ONNXRuntime/plugins/HiggsInteractionNetONNXJetTagsProducer.cc:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d, 858b30df39705ef42d7286f474ce5579d1d202ea
    - Deleted: 9bce4d38a9139531cb3c9fcbc796a99e57156519
  - DataFormats/BTauReco/interface/HiggsInteractionNetFeatures.h:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d
    - Deleted: 858b30df39705ef42d7286f474ce5579d1d202ea
  - RecoBTag/FeatureTools/plugins/HiggsInteractionNetTagInfoProducer.cc:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d, 1dee5d6b845f3d70d5f6ebf3f352eedeb10f63f2, 858b30df39705ef42d7286f474ce5579d1d202ea
    - Deleted: 9bce4d38a9139531cb3c9fcbc796a99e57156519
  - DataFormats/BTauReco/interface/HiggsInteractionNetTagInfo.h:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Deleted: 858b30df39705ef42d7286f474ce5579d1d202ea


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  PhysicsTools/PatAlgos/python/recoLayer0/bTagging_cff.py  modified in PR(s): #30016
  - File  RecoBTag/MXNet/plugins/BoostedJetMXNetJetTagsProducer.cc  modified in PR(s): #30508, #29934

jmduarte commented on RecoBTag/FeatureTools/interface/deep_helpers.h:
```diff
@@ -74,5 +74,38 @@ namespace btagbtvdeep {
+        return item->second;
+      } else {
+        throw cms::Exception("InvalidArgument") << "Cannot find variable info for " << name;
+      }
```
Changed as you suggest.

jmduarte commented on RecoBTag/FeatureTools/interface/deep_helpers.h:
```diff
@@ -74,5 +74,38 @@ namespace btagbtvdeep {
+    std::vector<std::string> var_names;
+    std::unordered_map<std::string, VarInfo> var_info_map;
+
+    VarInfo get_info(const std::string &name) const {
```
get prefix is now omitted.

cmsbuild commented:
Pull request #30072 was updated. @perrotta, @cmsbuild, @santocch, @slava77 can you please check and sign again.

slava77 commented on RecoBTag/ONNXRuntime/python/pfHiggsInteractionNet_cff.py:
```diff
@@ -0,0 +1,31 @@
+
+# nominal Higgs IN
+pfHiggsInteractionNetTags = _pfDeepBoostedJetTags.clone(
+    src = cms.InputTag('pfHiggsInteractionNetTagInfos'),
+    preprocessParams = pfHiggsInteractionNetPreprocessParams,
+    model_path = 'RecoBTag/Combined/data/HiggsInteractionNet/V00/IN.onnx',
+    flav_names = cms.vstring(
+        'probQCD',
+        'probHbb'
+    ),
```
```suggestion
    src = 'pfHiggsInteractionNetTagInfos',
    preprocessParams = pfHiggsInteractionNetPreprocessParams,
    model_path = 'RecoBTag/Combined/data/HiggsInteractionNet/V00/IN.onnx',
    flav_names = [ 'probQCD', 'probHbb' ]
    ),
```

jmduarte replied:
@slava77 Should I make this small change? Or wait?

slava77 replied:
that's my last remaining comment

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/RecoBTag-Combined#30

cmsbuild commented:
+1
Tested at: 5e7be345c0c644319c6fa6f15fdabd52b09db540
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/7653/summary.html
CMSSW: CMSSW_11_2_X_2020-07-02-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/7653/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3262 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2784120
 - DQMHistoTests: Total failures: 5
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2784064
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 36 files compared)
 - DQMHistoSizes: changed ( 10224.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 154 log files, 17 edm output root files, 37 DQM output files

jmduarte committed:
slava fix

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30072/16763

- This PR adds an extra 136KB to repository

- Found files with invalid states:
  - RecoBTag/ONNXRuntime/plugins/HiggsInteractionNetONNXJetTagsProducer.cc:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d, 858b30df39705ef42d7286f474ce5579d1d202ea
    - Deleted: 9bce4d38a9139531cb3c9fcbc796a99e57156519
  - DataFormats/BTauReco/interface/HiggsInteractionNetFeatures.h:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d
    - Deleted: 858b30df39705ef42d7286f474ce5579d1d202ea
  - RecoBTag/FeatureTools/plugins/HiggsInteractionNetTagInfoProducer.cc:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Modified: 5e2d8ea9b3472f504e2f6d5f91ce01103b69314d, 1dee5d6b845f3d70d5f6ebf3f352eedeb10f63f2, 858b30df39705ef42d7286f474ce5579d1d202ea
    - Deleted: 9bce4d38a9139531cb3c9fcbc796a99e57156519
  - DataFormats/BTauReco/interface/HiggsInteractionNetTagInfo.h:
    - Added: 9000e4502ec266f8de29946e4a56b660c205bb83
    - Deleted: 858b30df39705ef42d7286f474ce5579d1d202ea


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  PhysicsTools/PatAlgos/python/recoLayer0/bTagging_cff.py  modified in PR(s): #30016
  - File  RecoBTag/MXNet/plugins/BoostedJetMXNetJetTagsProducer.cc  modified in PR(s): #30508, #29934

cmsbuild commented:
Pull request #30072 was updated. @perrotta, @cmsbuild, @santocch, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-data/RecoBTag-Combined#30

cmsbuild commented:
-1

Tested at: 49d1b08190c5a9cfd648998931f0fa3eca48e907

CMSSW: CMSSW_11_2_X_2020-07-03-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/7688/summary.html

I found follow errors while testing this PR

**Failed tests:  AddOn**

* **AddOn**:

I found errors in the following addon tests:

Preparing to run ['cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run3_mc_PIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run3 --fileout file:RelVal_Raw_PIon_MC.root', 'cmsRun /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-07-03-2300/src/HLTrigger/Configuration/test/OnLine_HLT_PIon.py realData=False globalTag=@ inputFiles=@ ', 'cmsDriver.py RelVAddOnTest might have timed out: FAILED -  secs

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c596d4/7688/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3262 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2784120
 - DQMHistoTests: Total failures: 5
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2784064
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 36 files compared)
 - DQMHistoSizes: changed ( 10224.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 154 log files, 17 edm output root files, 37 DQM output files

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/30072	49d1b08 
- code changes are in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show differences only in the 
    - the addOn failure is unrelated (" RelVAddOnTest might have timed out: FAILED - secs")
- local test with 1K events in 136.88811 confirms that only the new two tags were added for pfHiggsInteractionNetTagsSlimmedAK8DeepTags probQCD and probHbb, while the existing tags for AK8 jets are unchanged
    - time per event is roughly consistent with the PR description, it adds up to about 0.8% per miniAOD workflow.
    - memory use increases by about 8 MiB (4.6 MiB constant per job, mostly in the model global cache, and 3.3 MiB per event in DeepBoostedJetONNXJetTagsProducer::produce ONNX runtime)

in the jetHt data 1K events wf 136.88811 the probHbb (a histogram in black; ignore the one in red) appears roughly as expected to show mostly background
![jet_origVStest30072_RunJetHT2018DreMINIAODULwf136p88811c_min2,max-2,patJets_slimmedJetsAK8__PAT_obj___pairDiscriVector__83__second84](https://user-images.githubusercontent.com/4676718/86684958-dfa0d000-bfb7-11ea-84b8-55c856199d43.png)

This PR should be merged together/after https://github.com/cms-data/RecoBTag-Combined/pull/30

silviodonato commented:
merge

Dr15Jones commented:
This pull request seems to be responsible for the 700+ failures in the RelVal IBs with the error

```
An exception of category 'ConfigFileReadError' occurred while
   [0] Processing the python configuration file named step3_RAW2DIGI_L1Reco_RECO_RECOSIM_EI_PAT.py
Exception Message:
 unknown python problem occurred.
RuntimeError: An exception of category 'FileInPathError' occurred.
Exception Message:
edm::FileInPath unable to find file RecoBTag/Combined/data/HiggsInteractionNet/V00/IN.onnx anywhere in the search path.
The search path is defined by: CMSSW_SEARCH_PATH
${CMSSW_SEARCH_PATH} is: /data/cmsbld/jenkins/workspace/ib-run-relvals/CMSSW_11_2_X_2020-07-07-2300/poison:/data/cmsbld/jenkins/workspace/ib-run-relvals/CMSSW_11_2_X_2020-07-07-2300/src:/data/cmsbld/jenkins/workspace/ib-run-relvals/CMSSW_11_2_X_2020-07-07-2300/external/slc7_amd64_gcc820/data:/cvmfs/cms-ib.cern.ch/nweek-02636/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-07-07-2300/src:/cvmfs/cms-ib.cern.ch/nweek-02636/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-07-07-2300/external/slc7_amd64_gcc820/data
Current directory is: /data/cmsbld/jenkins/workspace/ib-run-relvals/CMSSW_11_2_X_2020-07-07-2300/pyRelval/1.0_ProdMinBias+ProdMinBias+DIGIPROD1+RECOPROD1


At:
  /cvmfs/cms-ib.cern.ch/nweek-02636/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-07-07-2300/python/FWCore/ParameterSet/Types.py(808): insertInto
  /cvmfs/cms-ib.cern.ch/nweek-02636/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-07-07-2300/python/FWCore/ParameterSet/Mixins.py(373): insertContentsInto
  /cvmfs/cms-ib.cern.ch/nweek-02636/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-07-07-2300/python/FWCore/ParameterSet/Mixins.py(502): insertInto
  /cvmfs/cms-ib.cern.ch/nweek-02636/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-07-07-2300/python/FWCore/ParameterSet/Modules.py(162): insertInto
  /cvmfs/cms-ib.cern.ch/nweek-02636/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-07-07-2300/python/FWCore/ParameterSet/Config.py(1100): _insertManyInto
  /cvmfs/cms-ib.cern.ch/nweek-02636/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-07-07-2300/python/FWCore/ParameterSet/Config.py(1315): fillProcessDesc
  <string>(2): <module>


```

slava77 commented:
> This pull request seems to be responsible for the 700+ failures in the RelVal IBs with the error

it's not the PR, it's the incomplete merge of the external that's the issue

santocch commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (but tests are reportedly failing).
